pennylane/compiler/qjit_api.py

```diff
@@ -537,6 +541,41 @@ def _decorator(body_fn):
     return _decorator
 
 
+@functools.lru_cache
+def _get_for_loop_qfunc_prim():
+    """Get the loop_for primitive for quantum functions."""
+
+    import jax  # pylint: disable=import-outside-toplevel
+
+    for_loop_prim = jax.core.Primitive("for_loop")
+    for_loop_prim.multiple_results = True
+
+    @for_loop_prim.def_impl
+    def _(lower_bound, upper_bound, step, jaxpr_consts, *init_state, jaxpr_body_fn):
+
+        args = init_state
+        fn_res = args if len(args) > 1 else args[0] if len(args) == 1 else None
+
+        for i in range(lower_bound, upper_bound, step):
+
+            tmp_args = (i, *args)
+            fn_res = jax.core.eval_jaxpr(jaxpr_body_fn.jaxpr, jaxpr_consts, *tmp_args)
+            fn_res = jax.numpy.array(fn_res)
+            args = fn_res if len(args) > 1 else (fn_res,) if len(args) == 1 else ()
```

Comment:
So we know that `jax.core.eval_jaxpr` *always* returns a list of tensorlikes. So I'm little unsure what this line is doing.

What cases give rise to each logical branch?

Since `fn_res` is already a list, why do we want `([a, ], )`?

Reply:
It might be worth breaking this out into multi-line form:
```
if len(args) > 1:
    args = fn_res
elif len(args) == 1:
    args = (fn_res,)
else:
   args = ()
```

And making sure we actually have test coverage for each line.